gui/wxpython/gui_core/wrap.py

```diff
@@ -178,6 +179,42 @@ def SetToolTip(self, tip):
             wx.Button.SetToolTipString(self, tip)
 
 
+class ClearButton(Button):
+    """Wrapper around a Button with stock id wx.ID_CLEAR,
+    to disable default key binding on certain platforms"""
+    def __init__(self, *args, **kwargs):
+        Button.__init__(self, *args, **kwargs)
+        self.SetId(wx.ID_CLEAR)
+        if sys.platform == "darwin":
+            self.SetLabel(_("Clear"))
+        else:
+            self.SetLabel(_("&Clear"))
+
+
+class CancelButton(Button):
+    """Wrapper around a Button with stock id wx.ID_CANCEL, to disable
+    default key binding on certain platforms/wxpython versions"""
+    def __init__(self, *args, **kwargs):
+        Button.__init__(self, *args, **kwargs)
+        self.SetId(wx.ID_CANCEL)
+        if sys.platform == "darwin" and not CheckWxVersion([4, 1, 0]):
```

Comment:
Why special checks for wx 4.1.0 in CancelButton and not in other buttons?

Reply:
This particular issue with `Cancel` was addressed with https://github.com/wxWidgets/wxWidgets/pull/1131 which went to wxWidgets 3.1.4 (and thus wxPython 4.1). This check corrects this for wxPython <4.1 (and restores default label `&Cancel` for later versions).